### PR TITLE
Modified mbedk64f target 

### PR DESCRIPTION
**Pull request by polaroi8d**
As the mbed is a cpp project from which several classes were used we need to have .cpp files. 

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

### CODE REVIEW

**Comment by polaroi8d** [comment]
There are some problems with the renamed files, because if we rewrite their names, didn't find specific libaries, like cstddef.

**Comment by zherczeg** [comment]
LGTM. 
Just one more question: I am surprised that c -> cpp conversions do not cause any linking problems. Is it because of the extern C in api headers?

File: targets/mbedk64f/source/jerry_extapi.cpp [hunk 1]
```diff
@@ -14,19 +14,15 @@
  */
 
 #include "jerry-core/jerry.h"
-
 #include <stdlib.h>
 #include <stdio.h>
-
```
Comment (LaszloLango) [inline targets/mbedk64f/source/jerry_extapi.cpp]: These two empty lines help to separate the include types, such as jerry API, system and target headers. Why did you remove them?

File: targets/mbedk64f/source/jerry_run.cpp [hunk 2]
```diff
@@ -35,7 +36,7 @@ int js_entry (const char *source_p, const size_t source_size)
 
   js_register_functions ();
 
-  if (!jerry_parse (jerry_src, source_size))
+  if (!jerry_parse (jerry_src, source_size, &err_obj_p))
   {
     printf ("Error: jerry_parse failed\r\n");
     ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
```
Comment (LaszloLango) [inline targets/mbedk64f/source/jerry_run.cpp]: 'err_obj_p' must be released here.

File: targets/mbedk64f/source/jerry_run.cpp [hunk 3]
```diff
@@ -44,7 +45,7 @@ int js_entry (const char *source_p, const size_t source_size)
   {
     if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
     {
-      ret_code = jerry_run ();
+      ret_code = jerry_run (&err_obj_p);
```
Comment (LaszloLango) [inline targets/mbedk64f/source/jerry_run.cpp]: Please check `err_obj_p` and release if it is not `NULL`

**Comment by polaroi8d** [comment]
@zherczeg I guess.

File: targets/mbedk64f/source/jerry_run.cpp (hunk 3 above)
Comment (polaroi8d) [inline targets/mbedk64f/source/jerry_run.cpp]: Okey.

**Comment by akiss77** [comment]
@polaroi8d I've taken a closer look at the 4 open mbed-related PRs (this one + #981, #986, #989). They are way too much clones of each other. Except for differences in style (whitespace, comments, order of includes), the `js` and `source` directories are copy-paste clones of each other (perhaps the only difference between them is the serial port baud rate set in `main.cpp`). The real differences seem to be "only" in readme, yotta, module, and make files in the target root directory.

I'd propose unifying the mbed-related targets to prevent duplications. I could imagine a structure like
```
targets/
    mbed/
        js/
        source/
        port/
        board/
            k64f/
            stm32f401nucleo/
            stm32f4/
            stm32f429i/
```
Since the k64f support is already in the code base, I propose to update this PR to lay the foundations of the new structure as well as update the code to latest jerry (please note that an external port implementation is needed as identified in #1032).

Do have some discussion with @knightburton to make sure that the other PRs are aligned.

**Comment by polaroi8d** [comment]
@akiss77 there is a new PR #1019 where we already done, what you describe.

**Comment by akiss77** [comment]
@polaroi8d Ahm, working my way up from the bottom of the list, I did not get that far. My bad. However, if these four PRs are superceded by #1019 then could you please close these with an appropriate comment so they don't show up anymore as ones waiting for review?

**Comment by polaroi8d** [comment]
@akiss77 Of course.

**Comment by polaroi8d** [comment]
In this and some other PRs what we did, they are way too much clones of each other, so this is the reason why I'm closed this pull request, because we created a new PR ( #1019 ) where merged the mbed releated targets into one soruce.
